pkg/commands/git_commands/commit_loader.go

```diff
@@ -364,23 +358,40 @@ func (self *CommitLoader) setCommitMergedStatuses(refName string, commits []*mod
 			commits[i].Status = models.StatusMerged
 		}
 	}
-	return commits, nil
+	return commits
 }
 
-func (self *CommitLoader) getMergeBase(refName string) (string, error) {
-	info, err := self.getCurrentBranchInfo()
-	if err != nil {
-		return "", err
-	}
+func (self *CommitLoader) getMergeBase(refName string) string {
+	quotedBaseBranches := strings.Join(
+		lo.FilterMap(self.UserConfig.Git.BaseBranches, func(branch string, _ int) (string, bool) {
+			// We need to filter out base branches that don't exist, as they
+			// would make the git merge-base call fail with an error
+			if !self.branchExists(branch) {
+				return "", false
+			}
 
-	baseBranch := "master"
-	if strings.HasPrefix(info.RefName, "feature/") {
-		baseBranch = "develop"
+			return self.cmd.Quote(branch), true
+		}), " ")
+
+	if quotedBaseBranches == "" {
+		return ""
 	}
 
 	// swallowing error because it's not a big deal; probably because there are no commits yet
-	output, _ := self.cmd.New(fmt.Sprintf("git merge-base %s %s", self.cmd.Quote(refName), self.cmd.Quote(baseBranch))).DontLog().RunWithOutput()
-	return ignoringWarnings(output), nil
+	output, _ := self.cmd.New(fmt.Sprintf("git merge-base %s %s", self.cmd.Quote(refName), quotedBaseBranches)).DontLog().RunWithOutput()
```

Comment:
If we need to make N+1 calls to git, I think we should go through the list and find the first branch that exists and then use that alone in the merge-base. I prefer that because:
* it spares unnecessary calls to git due to the ability to early-exit
* it's logic that's easy to explain to a user

Reply:
That doesn't work. You can have a repo with several main branches (e.g. "master", "develop", "0.38.x-hotfixes"), and then each of your feature branches can be against a different one of those. We have that at work, some teams work against master, others against "next-major-version". So you'd need a per-branch config for which main branch this particular feature branch is based off of. (Side note: it would be nice if git itself would know this information somehow, which would solve the issue of accidentally creating github PRs against the wrong base branch, which happens all the time. But it doesn't.)

The beauty of the approach of this PR is that you only need to configure the set of possible base branches once, and then it just works for all feature branches, no matter which of these base branches it is based on.

Reply:
Couldn't that be misleading though? i.e. if I don't care about the develop branch but I have it checked out and it happens to be closer to my branch than master, I might think that I'm closer to master than I am. That seems like unexpected behaviour that could confuse people.

If we made it really easy to specify what the base branch of a given branch is (either via the config with some pattern matching or via the UI by manually saying 'branch X is based on branch Y') then I think users would prefer the more predictable behaviour of only ever considering a single base branch. What do you think?

Reply:
I don't think this can happen in practice (the `and it happens to be closer to my branch than master` bit, I mean). I'm happy to play through a few real-world examples if you want, but I think this is really safe.

I certainly don't want to configure the base branch for each of my feature branches. Pattern matching doesn't work, because the branch naming conventions are made by the company and I may not have the power to change them. And configuring it manually for each branch is unacceptable. Also, you'd have to change the config when you change the base branch -- sometimes I have a master-based branch and I want to rebase it onto "devel" just to see how many conflicts I would get.

My approach works in all these cases automatically.

Reply:
To be clear, with the pattern matching I was thinking that would be per-repo. And with manually configuring, I wasn't thinking of updating the config file, I was thinking of specifying the base branch within lazygit. I'm working on a branch right now that lets us write to the lazygit config while preserving existing comments and things so that's what I have in mind.

But as you say, in practice there may be no issue with considering the whole set of base branches. I'm open to implementing your approach and seeing if people trip up on it. At least if we find that users do get confused, we can change the meaning of the config without having to change the actual structure of it (which would be an annoying dance of deprecation).

I'll run this locally and see how it feels